src/metpy/plots/mapping.py

```diff
@@ -43,18 +43,28 @@ def _map_arg_names(source, mapping):
         return {cartopy_name: source[cf_name] for cartopy_name, cf_name in mapping
                 if cf_name in source}
 
+    ## issue 1844 deals with this block - changes made by lbunting
     @property
     def cartopy_globe(self):
         """Initialize a `cartopy.crs.Globe` from the metadata."""
         if 'earth_radius' in self._attrs:
+          #  kwargs['ellipse'] = 'sphere'
+          #  kwargs['semimajor_axis']=self._attrs['earth_radius']
+          #  kwargs['semiminor_axis']=self._attrs['earth_radius']
+            
             kwargs = {'ellipse': 'sphere', 'semimajor_axis': self._attrs['earth_radius'],
                       'semiminor_axis': self._attrs['earth_radius']}
+
         else:
             attr_mapping = [('semimajor_axis', 'semi_major_axis'),
                             ('semiminor_axis', 'semi_minor_axis'),
                             ('inverse_flattening', 'inverse_flattening')]
             kwargs = self._map_arg_names(self._attrs, attr_mapping)
-
+        #kwargs.update(self._map_arg_names(self._attrs, attr_mapping))
+            
+        if kwargs.get('inverse_flattening', None) == 0:
+            kwargs['ellipse'] = 'sphere'
+            kwargs.pop('inverse_flattening', None)
             # WGS84 with semi_major==semi_minor is NOT the same as spherical Earth
             # Also need to handle the case where we're not given any spheroid
             kwargs['ellipse'] = None if kwargs else 'sphere'
```

Comment:
Ok, this is the source of the current test failure. You should move this line (and it's comment) up to line 63 (in the `else`) to have it say:
```python
            # Override CartoPy's default ellipse setting depending on whether we have any metadata to map about the
            # spheroid.
            kwargs['ellipse'] = None if kwargs else 'sphere'